CASMINST-5307 (#2761)
Cray-HPE/docs-csm: operations/utility_storage/Troubleshoot_Ceph_MDS_reporting_slow_requests_and_failure_on_client.md

   > **`NOTE`** Messages such as "WARNING: The same type, major and minor should not be used for multiple devices" can be ignored. There is an upstream bug to address this issue.

> 4. (`ceph#`) Dump in-flight ops from the active mds.

Reviewer: ```suggestion
4. (`ceph#`) Dump in-flight ops from the active MDS.
```

   1. Find the active MDS.

      ```bash
      export active_mds=$(ceph fs status -f json-pretty|jq -r '.mdsmap[]|select(.state=="active")|.name')
      echo $active_mds
      ```

      Example output:

      ```text
      cephfs.ncn-s003.earesy
      ```

   2. Dump `ops_in_flight`.

   ```bash

>    ceph daemon mds.$mds dump_ops_in_flight

Reviewer 2: Shouldn't this be $active_mds?

   1. Usually the last "event" will have been an attempt to gather locks, or sending the operation off to the MDS log.

   1. If it is waiting on the OSDs, fix them.

>    1. If operations are stuck on a specific `inode`, you probably have a client holding caps which prevent others from using it.  Either:

Reviewer: ```suggestion
   1. If operations are stuck on a specific `inode`, then there is likely a client holding caps which prevent others from using it. This is caused by other of the following:
```

>       1. because the client is trying to flush out dirty data.

Reviewer: ```suggestion
      1. The client is trying to flush out dirty data.
```

>       1. because you have encountered a bug in CephFS' distributed file lock code (the file "capabilities" ["caps"] system).

Reviewer: ```suggestion
      1. There is a bug in CephFS' distributed file lock code (the file "capabilities" ["caps"] system).
```
